spec/inverted-index-test.js

```diff
@@ -0,0 +1,200 @@
+ const docs       = require('./books');
+ const docs_2     = require('./books_2');
+ const emptyDoc   = require('./empty.json');
+ const invalidDoc = require('./invalid.json');
+//A test suite to read data from books
+describe('Test suite for Inverted Index', () => {
+  //create an object of the Inverted Index class
+  const theIndex = new InvertedIndex();
+  //const emptyDoc = [];
+  const sentence        = "This# is@ a Sample *sentence &with some some ^dirt";
+  const multSearchTerms = "An Alice man tormented market".split(' ');
+  const allSearchTerms  = "Powerful dangerous young spy wizard elf".split(' ');
+
+  //create inverted indexes from the filenames
+  theIndex.createIndex('books.json', docs);
+  theIndex.createIndex('books_2.json', docs_2);
+  theIndex.createIndex('empty_doc.json', emptyDoc);
+  theIndex.createIndex('invalid_doc.json', invalidDoc);
+
+  /*
+  * Object that stores all the indexes created for every file
+  * It ensures that a new index does not overwrite the current index
+  * key -> String that is the name of the file
+  * value -> Inverted Index for that file
+  */
+  let allIndexes = theIndex.getIndex();
+
+  //create an Inverted Index for the file
+  theIndex.getIndex(docs);
+
+  describe('Inverted Index Class', () => {
+    it('Should be an instance of a class', () => {
+      expect(theIndex instanceof InvertedIndex).toBe(true);
+      expect(theIndex instanceof Object).toBe(true);
+      expect(typeof theIndex).toBe('object');
+    });
+  });
+
+  describe('Test for the cleaning function', () => {
+    it('Should return an array of strings', () => {
+      expect(theIndex.clean(sentence).constructor).toBe(Array);
```

Comment:
Your assertion does not efficiently test that the cleaning function is actually doing some cleaning like it should. You're only asserting that it should return an array.